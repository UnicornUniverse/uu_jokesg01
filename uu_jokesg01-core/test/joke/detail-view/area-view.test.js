import { render } from "../../tools";
import AreaView from "../../../src/joke/detail-view/area-view.js";

const mockContent = jest.fn(() => null);
jest.mock("../../../src/joke/detail-view/content.js", () => (props) => mockContent(props));

function getDefaultProps() {
  const jokeDataObject = {
    state: "ready",
    data: {
      id: "123",
      name: "Test joke",
      visibility: false,
      text: "Best joke ever",
      imageUrl: "https://via.placeholder.com/300x300",
      image: "1234",
      ratingCount: 5,
      averageRating: 2.5,
      categoryIdList: ["1", "2"],
      uuIdentity: "9-9",
      uuIdentityName: "Test User",
      sys: {
        cts: "2020-01-01T00:00:00.000Z",
      },
    },
  };

  const jokesPermission = {
    joke: {
      canAddRating: () => true,
      canManage: () => true,
    },
  };

  const jokesDataObject = {
    state: "ready",
    data: {
      categoryList: [
        { id: "1", name: "Category 1" },
        { id: "2", name: "Category 2" },
        { id: "3", name: "Category 3" },
      ],
    },
  };

  const awscDataObject = {
    state: "ready",
    data: {
      data: {
        artifact: {
          uuAppWorkspaceUri: "",
        },
      },
    },
  };

  const preferenceDataObject = {
    state: "ready",
    data: {
      showCategories: true,
      showAuthor: true,
      showCreationTime: true,
      disableUserPreference: false,
    },
  };
  return { jokesDataObject, jokesPermission, jokeDataObject, awscDataObject, preferenceDataObject };
}

function setup(props = getDefaultProps()) {
  mockContent.mockClear();
  render(<AreaView {...props} />);
  return { props };
}

describe(`UuJokesCore.Joke.DetailView.AreaView`, () => {
  it(`passes right data to Content`, async () => {
    setup();

    expect(mockContent).toBeCalledTimes(1);
    expect(mockContent.mock.lastCall[0]).toMatchSnapshot();
  });

  it(`logs preference loading error`, async () => {
    const defaultProps = getDefaultProps();

    const preferenceDataObject = {
      state: "readyNoData",
      data: null,
      handlerMap: {
        load: jest.fn(() => {
          throw new Error("Test error :-)");
        }),
      },
    };

    const props = { ...defaultProps, preferenceDataObject };
    AreaView.logger.error = jest.fn();

    setup(props);

    expect(AreaView.logger.error.mock.lastCall[0]).toMatchSnapshot();
  });
});
