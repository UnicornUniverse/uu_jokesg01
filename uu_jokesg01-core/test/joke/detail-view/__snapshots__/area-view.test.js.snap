// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UuJokesCore.Joke.DetailView.AreaView logs preference loading error 1`] = `"Error loading preference data"`;

exports[`UuJokesCore.Joke.DetailView.AreaView passes right data to Content 1`] = `
Object {
  "jokeDataObject": Object {
    "data": Object {
      "averageRating": 2.5,
      "categoryIdList": Array [
        "1",
        "2",
      ],
      "categoryList": Array [
        Object {
          "id": "1",
          "name": "Category 1",
        },
        Object {
          "id": "2",
          "name": "Category 2",
        },
      ],
      "id": "123",
      "image": "1234",
      "imageUrl": "https://via.placeholder.com/300x300",
      "name": "Test joke",
      "ratingCount": 5,
      "sys": Object {
        "cts": "2020-01-01T00:00:00.000Z",
      },
      "text": "Best joke ever",
      "uuIdentity": "9-9",
      "uuIdentityName": "Test User",
      "visibility": false,
    },
    "state": "ready",
  },
  "jokesDataObject": Object {
    "data": Object {},
    "state": "ready",
  },
  "jokesPermission": Object {
    "joke": Object {
      "canAddRating": [Function],
      "canManage": [Function],
    },
  },
  "nestingLevel": undefined,
  "parentStyle": Object {
    "paddingBottom": 24,
    "paddingLeft": 24,
    "paddingRight": 24,
    "paddingTop": 24,
  },
  "preferenceDataObject": Object {
    "data": Object {
      "disableUserPreference": false,
      "showAuthor": true,
      "showCategories": true,
      "showCreationTime": true,
    },
    "state": "ready",
  },
  "showDelete": false,
}
`;
